examples/gaussian_datatypes/example_explicit.py

```diff
@@ -0,0 +1,124 @@
+# For further information on the license, see the LICENSE.txt file.           #
+###############################################################################
+"""Simple DFT calculations with Gaussian Datatypes examples"""
+
+import os
+import sys
+
+import click
+import ase.io
+
+from aiida.engine import run
+from aiida.orm import (Code, Dict, StructureData)
+from aiida.common import NotExistent
+
+from gdt_data import load_data
+
+
+def example_gdt(cp2k_code):
+    """
+    Testing CP2K with the Basis Set and Pseudopotential stored in gaussian.basisset/pseudo.
+    In this example the names of the basissets and pseudopotentials are still explicitly named
+    in the CP2K input section. The plugin will then validate the names against the data objects
+    passed to ensure consistency, and alter the generated CP2K input file to include the parameters
+    contained in the specified basisset and pseudopotential objects, rather than relying on the
+    data being present on the target node.
+    Use this mode if you are migrating from implicitly tracked basissets and pseudos or if you
+    you have more complicated setups (e.g. having different basissets for the same kind on different sites).
+    """
+
+    thisdir = os.path.dirname(os.path.realpath(__file__))
+
+    # Structure.
+    structure = StructureData(ase=ase.io.read(os.path.join(thisdir, '..', "files", 'h2o.xyz')))
+
+    # in your actual code you may want to get the basisset and pseudopotential
+    # with a lookup in the database, here we can simply call load_data()
+    bsets, pseudos = load_data(prefix="MY-EXPLICIT-")
+    # in your code you will probably use something like:
+    #   bsets = [
+    #     BasisSet.get(element="H", name="DZVP-MOLOPT-GTH")
+    #     BasisSet.get(element="O", name="DZVP-MOLOPT-SR-GTH")
+    #   ]
+    # ...
+
+    # parameters
+    parameters = Dict(
+        dict={
+            "FORCE_EVAL": {
+                "METHOD": "Quickstep",
+                "DFT": {
+                    "QS": {
+                        "EPS_DEFAULT": 1.0e-12,
+                        "WF_INTERPOLATION": "ps",
+                        "EXTRAPOLATION_ORDER": 3,
+                    },
+                    "MGRID": {
+                        "NGRIDS": 4,
+                        "CUTOFF": 280,
+                        "REL_CUTOFF": 30
+                    },
+                    "XC": {
+                        "XC_FUNCTIONAL": {
+                            "_": "LDA"
+                        }
+                    },
+                    "POISSON": {
+                        "PERIODIC": "none",
+                        "PSOLVER": "MT"
+                    },
+                },
+                "SUBSYS": {
+                    "KIND": [
+                        {
+                            "_": "O",
+                            "POTENTIAL": "GTH {p.name}".format(p=pseudos["O"]),
+                            "BASIS_SET": "ORB {b.name}".format(b=bsets["O"]),
```

Comment:
@dev-zero do you know why in this particular case `GTH` and `ORB` are required? I tried to remove them and I got the following error, when executing CP2K:

```
 *******************************************************************************
 *   ___                                                                       *
 *  /   \                                                                      *
 * [ABORT]                                                                     *
 *  \___/      An invalid potential type <MY-EXPLICIT-AUTO-GTH-PADE-q6> was    *
 *    |                      specified for the atomic kind <O                  *
 *  O/|                                                                        *
 * /| |                                                                        *
 * / \                                                    qs_kind_types.F:2115 *
 *******************************************************************************
```

I am a bit confused, because in other examples the type specification is not required:

https://github.com/aiidateam/aiida-cp2k/blob/develop/examples/single_calculations/example_dft.py#L66-L77

Reply:
Yes, if the TYPE is missing (as in "POTENTIAL MY-FUNKY-PSEUDO"), CP2K tries to guess the type of the POTENTIAL by looking at the first segment of the name ("GTH-PADE" is interpreted as a GTH potential), hence you have to specify it explicitly.

Reply:
ok, thanks!